migrations/mysql/20240315194839_accounts_table.sql

```diff
@@ -1,11 +1,27 @@
 CREATE TABLE accounts(
-    id VARCHAR(100) NOT NULL,
-    PRIMARY KEY (id),
-    email VARCHAR(320) NOT NULL UNIQUE,
+    id VARCHAR(100) NOT NULL PRIMARY KEY,
     account_name VARCHAR(100) NOT NULL,
+    email VARCHAR(320),
     display_name VARCHAR(200) NOT NULL,
+    category ENUM('user', 'organization') NOT NULL,
     origin ENUM('cli', 'github') NOT NULL,
-    registered_at TIMESTAMP(6) NOT NULL
+    avatar_url VARCHAR(1000),
+    registered_at TIMESTAMP(6) NOT NULL,
+    is_admin TINYINT NOT NULL
 );
 
-CREATE INDEX accounts_name_idx ON accounts (account_name);
+CREATE UNIQUE INDEX idx_accounts_name ON accounts(account_name);
+CREATE UNIQUE INDEX idx_accounts_email ON accounts(email);
+
+CREATE TABLE accounts_github(
```

Comment:
Have you considered extending `accounts` table with:
- `provider` (e.g. "github")
- `provider_id` (varchar)

as opposed to having a separate table per provider?

I think as we add other auth providers like Google, Microsoft, LDAP etc. it would suck to create tables for each. Ideally adding another provider plugin would not affect DB migrations.

Reply:
What makes you think there will be always a single identifier in every provider? Do you know how account reference looks in every possible protocol?

I kind of enjoy strict typing and database-level validation, which current structure introduces.

Reply:
I'd be really surprised to see a provider with multiple identifiers, but OK, let's see how this model fares.

Reply:
1) LDAP needs at least 2 fields, as far as I remember: user identifier + particular server address. There are also some groups, sort of namespaces.

2) Who said it will be limited to identifers. This design allows saving any number of provider-specific fields associated with an account. We might want to save other type of GitHub properties to customize the look of UI, if needed.